.github/workflow/publish.yml

```diff
@@ -0,0 +1,37 @@
+# .github/workflows/publish.yml
+name: Publish to NPM
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-and-publish:
```

Comment:
I think deploying on any push in master could work, if we don't push there anything beside PRs, however it has few limitations, it cannot be used in pull request builds (because it will push each time), it does push a tag in the git, and does not create a release automatically that would link with it.

Usually, we do build and publish jobs separately, that triggers on any commit for building job, and on draft new release for the publish one.